pkg/testing/kubernetes/kind/provisioner.go

```diff
@@ -0,0 +1,276 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package kind
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/testing/runner"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/e2e-framework/klient"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+)
+
+const (
+	Name = "kind"
+)
+
+const clusterCfg string = `
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  kubeadmConfigPatches:
+  - |
+    kind: ClusterConfiguration
+    scheduler:
+      extraArgs:
+        bind-address: "0.0.0.0"
+        secure-port: "10259"
+    controllerManager:
+      extraArgs:
+        bind-address: "0.0.0.0"
+        secure-port: "10257"
+`
+
+func NewProvisioner(version string) runner.InstanceProvisioner {
+	return &provisioner{version: version}
+}
+
+type provisioner struct {
+	logger  runner.Logger
+	version string
+}
+
+func (p *provisioner) Name() string {
+	return Name
+}
+
+func (p *provisioner) Type() runner.ProvisionerType {
+	return runner.ProvisionerTypeK8SCluster
+}
+
+func (p *provisioner) SetLogger(l runner.Logger) {
+	p.logger = l
+}
+
+func (p *provisioner) Supported(batch define.OS) bool {
+	return batch.Type == define.Kubernetes && batch.Arch == runtime.GOARCH && (batch.Distro == "" || batch.Distro == "kind")
+}
+
+func (p *provisioner) Provision(ctx context.Context, cfg runner.Config, batches []runner.OSBatch) ([]runner.Instance, error) {
+	if len(batches) == 0 {
+		return nil, errors.New("kind provisioner only supports a single batch")
```

Comment:
I does support concurrent clusters but two thoughts I had where the following:
- how excessive/feasible will be in terms of resources on the VM to provision at the same time more than one k8s clusters
- how to pass down to each test invocation the appropriate KUBECONFIG so that the info.KubeClient will target the correct k8s cluster

🤔

Reply:
We support multiple with multipass, it is probably true that at the moment we don't need to support multiple batches. I am okay with this at the moment. Just didn't know if it was a restriction of `kind` to not allow more than 1 at a time.

Reply:
I went ahead with supporting multiple cluster is somebody wants to 🙂